sw/host/tests/chip/spi_device_tpm_test/src/main.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, Result};
+use clap::Parser;
+use regex::Regex;
+use std::path::PathBuf;
+use std::time::Duration;
+
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::execute_test;
+use opentitanlib::test_utils::init::InitializeTest;
+use opentitanlib::tpm::{self, Driver};
+use opentitanlib::uart::console::{ExitStatus, UartConsole};
+
+#[derive(Debug, Parser)]
+struct Opts {
+    #[command(flatten)]
+    init: InitializeTest,
+
+    /// Console receive timeout.
+    #[arg(long, value_parser = humantime::parse_duration, default_value = "600s")]
+    timeout: Duration,
+
+    /// Name of the SPI interface to connect to the OTTF console.
+    #[arg(long, default_value = "TPM")]
+    spi: String,
+
+    /// Path to the firmware's ELF file, for querying symbol addresses.
+    #[arg(value_name = "FIRMWARE_ELF")]
+    firmware_elf: PathBuf,
+}
+
+fn tpm_read_test(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
+    let uart = transport.uart("console")?;
+    let spi = transport.spi(&opts.spi)?;
+    transport.pin_strapping("SPI_TPM")?.apply()?;
```

Comment:
Stupid question: I would have expected the console name and the strapping name to be related somehow. Or is `SPI_TPM` a strapping for **all** SPIs?

Reply:
This has nothing to do with the console, actually. It's a strapping for the debugger to set itself up to drive the TPM portion of spi_device on earlgrey, which has a separate GPIO for the chip select and shares all the other pins with the BOOTSTRAP SPI configuration.

The BOOTSTRAP and TPM SPI devices cannot be used simultaneously. On hyperdebug, they are actually driven by entirely different IPs: There is no one SPI IP in that MCU that can handle both use cases.

The strapping config is here: https://github.com/lowRISC/opentitan/blob/c8de39a70f38a46c9546ae918798efeb49e03d67/sw/host/opentitanlib/src/app/config/hyperdebug_cw310.json#L44-L76